### ISSUE TITLE
SLiC-HF: Sequence Likelihood Calibration with Human Feedback

### ISSUE THREAD

**Issue by ooooohira** [ooooohira/paper_summary#43]
### 論文リンク
[https://arxiv.org/abs/2305.10425](https://arxiv.org/abs/2305.10425)

### 著者
Yao Zhao, Rishabh Joshi, Tianqi Liu, Misha Khalman, Mohammad Saleh, Peter J. Liu

### 投稿年
Wed, 17 May 2023

## 新規性や利点
RLHFに代わりうるような手法であるSLiC-HFを考案した。

## 提案手法
<img width="1327" alt="スクリーンショット 2024-05-17 13 53 16" src="https://github.com/ooooohira/paper_summary/assets/131585401/bd5e549d-a905-440d-9102-ac34003eb6b2">
報酬モデル(2種で、ランキングベースか、pointベースか、T5-XXL (11B))を学習させ、それのフィードバックを用いるか。
それとも既存のオフラインの嗜好データをそのまま用いるかの3通りを試した。
yrefには、pointベースのT5-XXL (11B)を用いるか、SFTデータのreferenceを用いるかの2通りを試す。

## 実験
SLiC-HFとか、Continue Fine-tuning では、 Reddit TL;DR summarization datasetsのDsft(通常のSFTに使用するデータ)ではなく他のDsftデータのpromptを用いていると思われる。
色々な比較実験をした。

## 結果と考察
win rateはランキングベースか、pointベースか、T5-XXL (11B)を用いて、human referencesに対する勝率。
Results shows that our ranking model has accuracy of 73.23% on DHF validation, about 2% higher than our reward model which has accuracy of 71.34%でやや、ランキングベースの方がいい。

<img width="1219" alt="スクリーンショット 2024-05-17 14 29 55" src="https://github.com/ooooohira/paper_summary/assets/131585401/e9aa35e4-e40d-4033-997e-019e0c36f235">
regularizationによる違いはない→SLiC-HFの時にreferenceなしでもいける。

<img width="1298" alt="スクリーンショット 2024-05-17 15 36 01" src="https://github.com/ooooohira/paper_summary/assets/131585401/77824b53-88c1-4fe4-a34c-ba5b069cb57d">
人で評価でも、SLiC-HFは強い。

<img width="1293" alt="スクリーンショット 2024-05-17 15 43 57" src="https://github.com/ooooohira/paper_summary/assets/131585401/53c46ccb-cdd6-47e7-bd27-6e0642f3a875">
RLHFと比べると、ランキングベースの本手法だとやや良い結果が得られた。
また、生成している文の長さを考慮すると、原点対照でRLHFと似ているような結果が得られた。

the policy, value, reward, and SFT modelsを学習速度高速化のためにメモリに載せる必要があるPPOと比べると、本手法のrewardはメモリにのせる必要がないため、メモリ自体は1/4で済む。
学習に使うデータのデーコードは全て同じポリシーで行われるため並列に高速に学習に使うデータをデコードとranking付けできて(no decoding in the training loop)、バッチごとにポリシーが変わってそこからデコードしrewardを導出するPPOよりかなり高速。

SLiC-HFは2つの要約の相対的な順位にのみ関心を持つため、このペアワイズからポイントワイズへのノイズは回避されます。このことがSLiC-HFに役立つと推測される。
